zerver/lib/markdown/__init__.py

```diff
@@ -2078,6 +2079,35 @@ def handleMatch(  # type: ignore[override] # supertype incompatible with supersu
         return None, None, None
 
 
+class EscapePostprocessor(markdown.postprocessors.UnescapePostprocessor):
+    """
+    This is a Zulip modification to the upstream.
+    Originally `UnescapePostprocessor` doesn't deal with escaping the characters.
+    Therefore, we change it to `EscapePostprocessor` which escapes char: `&`, `<`, `>`, `"`.
+    This modification ensures that Backslash Escaping doesn't introduce any XSS
+    issues, and we follow CommonMark.
+
+    It must be noted that the original behaviour of `UnescapePostprocessor` is left
+    unchanged, i.e. `Convert some escaped characters back from integers`. The only modified part
+    is that we escape the converted characters with the help of the function `char_escape`.
+    """
+
+    def unescape(self, m: Match[AnyStr]) -> str:
+        return self.char_escape(super().unescape(m))
```

Comment:
We don’t need a custom escape function.

```suggestion
        return html.escape(super().unescape(m))
```